ShopApp/ShopApp/Controller/ProductDetailsViewController.swift

```diff
@@ -110,144 +97,93 @@ class ProductDetailsViewController: UIViewController {
-                var isAdded = -1
-                var quantityForProduct = 0
-
-                for data in self.cart.keys where data.elementsEqual(name) {
-                    guard let sizeForProduct = cart[data] else { return }
-                    let size = sizeForProduct["size"]
-                    if sizeSelected == size {
-                        guard let quantity = sizeForProduct["quantity"] else { return }
-                        quantityForProduct =  Int(quantity) ?? 0
-                        isAdded = 0
-                    } else { isAdded = 1 }
-                }
-
-                if self.sizeSelected == "" {
-                    messagge = "First you have to choose a size!"
-                } else {
-                        guard let email = Auth.auth().currentUser?.email as? String else { return }
-                        switch isAdded {
-                        case -1:
-                            messagge = "Product added to cart!"
-                        case 1:
-                            messagge = "The size was changed!"
-                        case 0:
-                            messagge = "The quantity is changed!"
-                        default:
-                           messagge = "Error from add to cart"
-                        }
-
-                    try await Database.database().reference().child("users/" + email.replacingOccurrences(of: ".", with: " ") + "/cart/\(product.getName())").setValue(["size": sizeSelected, "quantity": "\(quantityForProduct + 1)"])
-                }
+    func favoritesAction() {
+        guard let email = Auth.auth().currentUser?.email as? String else { return }
+        let emailRef = email.replacingOccurrences(of: ".", with: " ")
 
-                let allFieldRequired = UIAlertController(title: "Message",
-                                                         message: messagge,
-                                                         preferredStyle: .alert)
-                allFieldRequired.addAction(UIAlertAction(title: String(localized: "Ok", comment: "Ok"), style: .cancel, handler: nil))
-                present(allFieldRequired, animated: true, completion: nil)
-            } catch {
-                print(error)
-            }
+        if isFavorite == false {
+            Database.database().reference().child("users/\(emailRef)/favourites/\(self.product.getName())").setValue("favorite")
+        } else {
+            Database.database().reference().child("users/\(emailRef)/favourites/\(self.product.getName())").removeValue()
         }
     }
 
-
-    // MARK: - Return favorites from user
-    func getFavourites() async -> [String] {
-        guard let email = Auth.auth().currentUser?.email as? String else { return [String]() }
+    // MARK: - Set the like button in the beginning
+    func setFavoritesButton() {
+        guard let email = Auth.auth().currentUser?.email as? String else { return }
         let ref = Database.database().reference()
```

Comment:
You can create 'ref' as a global constant since it is used in line 105, 107, 114, 132 and 160